operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -410,10 +411,21 @@ func extractS3ConfigSecret(s *corev1.Secret, credentialMode lokiv1.CredentialMod
 		roleArn  = s.Data[storage.KeyAWSRoleArn]
 		audience = s.Data[storage.KeyAWSAudience]
 		// Optional fields
-		region         = s.Data[storage.KeyAWSRegion]
-		forcePathStyle = !strings.HasSuffix(string(endpoint), awsEndpointSuffix)
+		region = s.Data[storage.KeyAWSRegion]
 	)
 
+	// Determine if we should use path style URLs for S3
+	// default to false for non-AWS endpoints
+	forcePathStyle := !strings.HasSuffix(string(endpoint), awsEndpointSuffix)
+	// Check if the user has specified force_path_style
+	if configForcePathStyle, ok := s.Data[storage.KeyAWSForcePathStyle]; ok {
+		strForcePathStyle := string(configForcePathStyle)
+		if strForcePathStyle != "true" && strForcePathStyle != "false" {
+			return nil, fmt.Errorf("%w: %s", errS3ForcePathStyleInvalid, storage.KeyAWSForcePathStyle)
+		}
+		forcePathStyle = string(configForcePathStyle) == "true"
```

Comment:
We can use the already-converted value here:

```suggestion
		forcePathStyle = strForcePathStyle == "true"
```